ui/app/hooks/useCurrencyDisplay.js

```diff
@@ -0,0 +1,69 @@
+import { useMemo } from 'react'
+import { useSelector } from 'react-redux'
+import { formatCurrency, getValueFromWeiHex } from '../helpers/utils/confirm-tx.util'
+
+/**
+ * Defines the shape of the options parameter for useCurrencyDisplay
+ * @typedef {Object} UseCurrencyOptions
+ * @property {string} [displayValue]     - When present is used in lieu of formatting the inputValue
+ * @property {string} [prefix]           - String to prepend to the final result
+ * @property {number} [numberOfDecimals] - Number of significant decimals to display
+ * @property {string} [denomination]     - Denomination (wei, gwei) to convert to for display
+ * @property {string} [currency]         - Currency type to convert to. Will override nativeCurrency
+ */
+
+/**
+ * Defines the return shape of the second value in the tuple
+ * @typedef {Object} CurrencyDisplayParts
+ * @property {string} [prefix]  - string to prepend to the value for display
+ * @property {string} value     - string representing the value, formatted for display
+ * @property {string} [suffix]  - string to append to the value for display
+ */
+
+/**
+ * useCurrencyDisplay hook
+ *
+ * Given a hexadecimal encoded value string and an object of parameters used for formatting the
+ * display, produce both a fully formed string and the pieces of that string used for displaying
+ * the currency to the user
+ * @param {string} inputValue          - The value to format for display
+ * @param {UseCurrencyOptions} opts    - An object for options to format the inputValue
+ * @return {[string, CurrencyDisplayParts]}
+ */
+export function useCurrencyDisplay (inputValue, { displayValue, prefix, numberOfDecimals, denomination, currency, ...opts }) {
+  const { currentCurrency, nativeCurrency, conversionRate } = useSelector(
+    ({ metamask: { currentCurrency, nativeCurrency, conversionRate } }) => ({
+      currentCurrency,
+      nativeCurrency,
+      conversionRate,
+    })
+  )
+
+  const toCurrency = currency || currentCurrency
+
+  const value = useMemo(() => {
+    if (displayValue) {
+      return displayValue
+    }
+    return formatCurrency(
+      getValueFromWeiHex({
+        value: inputValue,
+        fromCurrency: nativeCurrency,
+        toCurrency,
+        conversionRate,
+        numberOfDecimals: numberOfDecimals || 2,
+        toDenomination: denomination,
+      }),
+      toCurrency
+    )
+  }, [inputValue, nativeCurrency, conversionRate, displayValue, numberOfDecimals, denomination, toCurrency])
+
+  let suffix
+
+  if (!opts.hideLabel) {
+    suffix = opts.suffix || toCurrency.toUpperCase()
+  }
```

Comment:
The logic in the old container was a little different:

```suggestion
  let suffix = opts.suffix

  if (!suffix && !opts.hideLabel) {
    suffix = toCurrency.toUpperCase()
  }
```

Though... that doesn't really make any sense 🤔 `hideLabel` doesn't do anything in that case.

Your version seems a lot more sensible, but in order to change this, we'd need to remove the `hideLabel` prop from anywhere it's passed in alongside a `suffix`, so that the `suffix` is still shown correctly. I found one such place: `ui/app/pages/permissions-connect/choose-account/choose-account.component.js`, line 107. So maybe the better solution is to delete that one instance of `hideLabel` being passed in with a `suffix`.

Reply:
Removed the prop from the choose-account instance of UserPreferencedCurrencyDisplay, which resulted in `nativeCurrency` being unused in that component as well as the permissions-connect component and container so I had to touch a few more files but I removed it. I do think that this was the correct path to resolving.

Reply:
I'm a little confused - I meant to suggest removing `hideLabel`, not `suffix`.

Previously the suffix _was_ shown on that component, and the `hideLabel` prop was passed in despite it having no effect. I suggested it be removed because it never did anything to begin with, but it would result in the suffix being hidden with this change in logic you introduced.

Unless I'm mistaken about the suffix having been shown previously on that component?

Reply:
I think I was the confused one. Sorry for the churn.

Reply:
reverted that change and removed hideLabel. hideLabel was definitely the unused prop that needed excising.

Reply:
Thanks!